api/v1alpha1/lua_types.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// LuaFilter defines a Lua extension that applies globally
+// to all requests passing through the filter chain.
+type LuaFilter struct {
+	// SourceCodes is a map of named LuaSource codes that can be referenced in LuaPerRouteFilter
+	//
+	// +kubebuilder:validation:MaxProperties=16
+	// +optional
+	SourceCodes map[string]*LuaSource `json:"sourceCodes"`
+	// DefaultSourceCode is the default LuaSource that will be executed
+	// unless overridden by LuaPerRouteFilter configured for the selected route
+	DefaultSourceCode *LuaSource `json:"defaultSourceCode"`
+	// StatPrefix is the additional prefix to use when emitting statistics.
+	// By default, metrics are emitted in .lua. namespace.
+	// If multiple lua filters are configured in a filter chain, the stats from each filter instance can be emitted
+	// using custom stat prefix to distinguish emitted statistics.
+	//
+	// +optional
+	StatPrefix *string `json:"statPrefix,omitempty"`
+}
+
+// LuaPerRouteFilter defines a Lua extension that applies to specific routes based on route match.
+// Only one of Disabled, Name or OverrideSource must be set.
+// +kubebuilder:validation:XValidation:rule="has(self.disabled) ? (!has(self.name) && !has(self.overrideSource)) : (has(self.name) ? !has(self.overrideSource) : has(self.overrideSource))",message="Exactly one of disabled, name, or overrideSource must be set."
+type LuaPerRouteFilter struct {
```

Comment:
We should not need a `LuaPerRouteFilter` struct in the API. Let's define a universal API that can be used for both a Gateway or an HTTPRoute.